commands/tmplexec.go

```diff
@@ -77,7 +77,13 @@ type cmdExecFunc func(cmd string, args ...interface{}) (interface{}, error)
 // Returns 2 functions to execute commands in user or bot context with limited about of commands executed
 func TmplExecCmdFuncs(ctx *templates.Context, maxExec int, dryRun bool) (userCtxCommandExec cmdExecFunc, botCtxCommandExec cmdExecFunc) {
 	execUser := func(cmd string, args ...interface{}) (interface{}, error) {
-		mc := &discordgo.MessageCreate{ctx.Msg}
+		messageCopy := *ctx.Msg
+		if ctx.Msg.ChannelID != ctx.Msg.GuildID { //punishment dms shouldnt be able to execCC followed by exec/execAdmin to prevent loops
```

Comment:
Don't think this is needed, as the moderation commands only returns when the punish dm has been sent, can there can only be 1 instance of each command per user being executed

Reply:
Ah okay , removed the check, it is still necessary to check if CS is a nil pointer before accessing CS.ID so kept that check. It is nil pointer in some cases like direct exec/execAdmin within punishment dms ( i got a runtime panic when i was running without this condition earlier ).